src/editor/components/modals/file-modal/manage-file-sources/file-source-list-item.jsx

```diff
@@ -0,0 +1,245 @@
+import React from "react";
+import PropTypes from "prop-types";
+import styled from "react-emotion";
+import { connect } from "react-redux";
+import Delete from "@material-ui/icons/Delete";
+
+import { ListItem, ListMetadata } from "../../../../../shared/components/list";
+
+import DeleteModal from "../../../../../shared/components/delete-modal";
+
+import FileSourceListItemDescription from "./file-source-list-item-description";
+
+import InProgress from "./in-progress";
+
+import { DELETE_ANIMATION_LENGTH_MS } from "../shared/constants";
+
+import {
+  createFileUpdateOperation as createFileUpdateOperationAction,
+  deleteFileSource as deleteFileSourceAction,
+  setConfirmDeleteID as setConfirmDeleteIDAction,
+  setIsDeletingAnimationID as setIsDeletingAnimationIDAction
+} from "../../../../actions/file-source-actions";
+
+import {
+  TextButton,
+  OutlineButton
+} from "../../../../../shared/components/buttons";
+
+const FileSourceListItemContainer = styled(ListItem)`
+  transition: height ${DELETE_ANIMATION_LENGTH_MS}ms,
+    padding-top ${DELETE_ANIMATION_LENGTH_MS}ms,
+    padding-bottom ${DELETE_ANIMATION_LENGTH_MS}ms,
+    opacity ${DELETE_ANIMATION_LENGTH_MS}ms;
+  transition-timing-function: ease-out;
+
+  &.disappearing {
+    height: 0px;
+    padding-top: 0px;
+    padding-bottom: 0px;
+    overflow: hidden;
+    opacity: 0;
+  }
+`;
+
+const FileSourceInterval = styled(ListMetadata)`
+  font-size: 12px;
+  color: gray;
+  padding-left: 6px;
+  padding-right: 6px;
+  margin-left: 4px;
+  margin-right: 4px;
+  text-align: center;
+  line-height: 1.1;
+  min-width: 100px;
+  align-self: center;
+`;
+
+const DownloadNowButtonContainer = styled(ListMetadata)`
+  text-align: center;
+  position: relative;
+  height: 100%;
+  min-width: 150px;
+  display: grid;
+  margin: auto;
+  align-content: center;
+  overflow-y: hidden;
+  cursor: pointer;
+`;
+
+const DownloadNowButton = styled(OutlineButton)`
+  min-width: 150px;
+  padding-left: 3px;
+  padding-right: 0px;
+`;
+
+const DeleteButtonContainer = styled(ListMetadata)`
+  align-self: center;
+  display: grid;
+  align-content: center;
+
+  button {
+    display: grid;
+  }
+`;
+
+const DeleteButton = styled(TextButton)`
+  padding: 2px;
+  margin-left: 20px;
+`;
+
+export function FileSourceListItemUnconnected({
+  id,
+  url,
+  filename,
+  latestFileUpdateOperationStatus,
+  hasBeenRun,
+  updateInterval,
+  lastUpdated,
+  isCurrentlyRunning,
+  failureReason,
+  showFailureReason,
+  listSize,
+  isDeletingClass,
+  deleteFileSource,
+  createFileUpdateOperation,
+  isConfirmingDelete,
+  setConfirmDeleteID,
+  setIsDeletingAnimationID
+}) {
+  return (
+    <>
+      <DeleteModal
+        visible={isConfirmingDelete}
+        title={`Delete the file source for "${filename}"?`}
+        content="This action will not delete any files downloaded by the file source."
+        onCloseOrCancel={() => {}}
```

Comment:
FIXME: this got removed while I was refactoring. We need an `onCloseOrCancel` function to be defined here before this merges. WIll do this today.